stdlib/public/runtime/ImageInspectionCommon.cpp

```diff
@@ -28,9 +28,10 @@
 
 namespace swift {
 
+#ifndef NDEBUG
```

Comment:
This seems wrong; the `registered` field is always consulted to lookup the metadata sections.  I don't think that this should be predicated on an asserts build.

Reply:
This hasn't changed from the last PR. :)

Reply:
Talked to @compnerd off GitHub to clarify. We can probably remove `registered` entirely and use the runtime's knowledge, but we'll save that for a separate PR.